tests/kernel/userbufer_validate/src/userbuffer_validate.c

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <mmustructs.h>
+#include <arch/x86/arch.h>
+#include <linker/linker-defs.h>
+#include <ztest.h>
+
+void x86_mmu_set_pte_rw_bit(void *addr, bool value)
```

Comment:
Why do we have these functions here?
We already have _x86_mmu_set_flags(). Use that instead.
You can make these functions call _x86_mmu_set_flags() if you want. But you should not be poking the page tables directly in this file.